In [77]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score
import matplotlib.pyplot as plt

In [15]:
#importar datos
df = pd.read_csv('wbes_sintetico.csv')
df.head()

,employees,firm_age_years,training,access_finance,exporter,digital_payments,competition_level,sector,sales_annual_usd,labor_productivity_usd_per_employee,successful
0,13.0,10.0,1.0,1.0,0,1,Media,Manufactura,42879.327130,5374.714150,1
1,9.0,6.0,0.0,0.0,0,1,Baja,Servicios,54570.897174,4728.282631,0
2,15.0,10.0,0.0,0.0,1,1,Baja,Servicios,56311.260030,3104.627628,0
3,25.0,12.0,1.0,0.0,1,1,Baja,Servicios,93419.359675,3129.017055,1
4,9.0,9.0,0.0,0.0,0,0,Baja,Manufactura,24608.889361,2824.645056,1


## 1. Deleting extra columns

In [68]:
df_copy = df.drop(columns=['exporter', 'digital_payments', 'sector', 'labor_productivity_usd_per_employee', 'successful'])
df_copy.head()

,employees,firm_age_years,training,access_finance,competition_level,sales_annual_usd
0,13.0,10.0,1.0,1.0,Media,42879.327130
1,9.0,6.0,0.0,0.0,Baja,54570.897174
2,15.0,10.0,0.0,0.0,Baja,56311.260030
3,25.0,12.0,1.0,0.0,Baja,93419.359675
4,9.0,9.0,0.0,0.0,Baja,24608.889361


## 2. Checking for nulls and cleaning

In [69]:
df_copy.isna().sum()

employees            10
firm_age_years       26
training             33
access_finance       35
competition_level    33
sales_annual_usd     27
dtype: int64

In [70]:
df_clean = df_copy.dropna()
df_clean.isna().sum()

employees            0
firm_age_years       0
training             0
access_finance       0
competition_level    0
sales_annual_usd     0
dtype: int64

## 3. Variables independientes y variable dependiente

In [71]:
X = df_clean.drop(columns=['sales_annual_usd'])
Y = df_clean[['sales_annual_usd']] 

X_encoded = pd.get_dummies(
    X,
    columns=['competition_level'],
    drop_first=True
)


In [72]:
X_encoded

,employees,firm_age_years,training,access_finance,competition_level_Baja,competition_level_Media
0,13.0,10.0,1.0,1.0,False,True
1,9.0,6.0,0.0,0.0,True,False
2,15.0,10.0,0.0,0.0,True,False
3,25.0,12.0,1.0,0.0,True,False
4,9.0,9.0,0.0,0.0,True,False
...,...,...,...,...,...,...
1193,12.0,5.0,1.0,0.0,False,True
1195,13.0,0.0,0.0,0.0,False,True
1196,10.0,3.0,0.0,0.0,True,False
1198,11.0,10.0,0.0,1.0,False,True


## 4. Modelo

In [73]:
#Ajustar el modelo de regresión múltiple
model = LinearRegression() 
model.fit(X_encoded, Y) #Entrenando el modelo 
y_pred = model.predict(X_encoded)
#y_pred

## 5. Métricas

In [74]:
#Metricas estadísticas
mse = mean_squared_error(Y, y_pred) #Error de la media cuadrática
r2 = r2_score(Y, y_pred) #varianza

In [75]:
print('Valor de intersección : ', model.intercept_)
print('Coeficientes: ', model.coef_)
print('Error cuadrático medio', mse)
print('Varianza: ', r2)

Valor de intersección :  [10.77378748]
Coeficientes:  [[1156.78924837  603.59510095 6527.46524532 8476.20907451 7538.76005744
  4020.61292152]]
Error cuadrático medio 162254287.75167915
Varianza:  0.45557684466025605


## 6. Conclusiones
* Cada empleado adicional aumenta las ventas anuales en 1157 USD aprox
* Cada año adicional de operación aumenta ventas en 603 USD aprox
* Empresas con capacitación venden 6527 USD aprox. más que las que no
* Acceso a financiamiento está fuertemente asociado a mayores ventas
* Baja competencia = mayores ventas, comparadas con Alta competencia
* Competencia media = más ventas que alta, pero menos que baja

Los resultados del modelo son consistentes con la teoría económica, mostrando que el tamaño de la empresa, la experiencia, el acceso a financiamiento y un menor nivel de competencia están asociados positivamente con mayores ventas anuales.